File: Programmers/src/Javascript/다항식더하기.js
function solution(polynomial) {
    let arr = polynomial.split(" + ");
    let numx = 0, num = 0;
    
    for(let i=0; i<arr.length; i++){
        if(arr[i].includes('x')){
            let tmp = arr[i].split("x");
            
            if(tmp[0]==="")
                numx+=1;
            else
                numx+=Number(tmp[0]);
        } else {
            num += Number(arr[i]);
        }
    }
    
    if(numx !== 0 && num !== 0)
        return numx === 1 ? `x + ${num}` : `${numx}x + ${num}`;
    else if(numx !== 0 && num === 0)
        return numx === 1 ? 'x' : `${numx}x`;
    else if(numx === 0 && num !==0)
        return `${num}`;
    else
        return "0";
}

File: Programmers/src/Java/Lv0/다항식 더하기.java
class Solution {
    public String solution(String polynomial) {
        String answer = "";
        String[] arr = polynomial.split(" +");
        int sum1 = 0, sum2 = 0;
        
        for(String s : arr) {
            if(s.equals("x"))
                sum1 += 1;
            else if(s.contains("x"))
                sum1 += Integer.parseInt(s.substring(0, s.length()-1));
            else if(!s.equals("+"))
                sum2 += Integer.parseInt(s);
        }
        
        if(sum1!=0 && sum2==0) {
            if(sum1==1)
                answer += "x";
            else
                answer += sum1 + "x";
        }
        
        if(sum1!=0 && sum2!=0) {
            if(sum1==1)
                answer += "x" + " + " + sum2;
            else
                answer += sum1 + "x" + " + " + sum2; 
        }
        
        if(sum1==0 && sum2!=0)
            answer += sum2;
        
        return answer;
    }
}